File: src/main/java/br/com/dusty/dservices/util/NTPUtils.java
package br.com.dusty.dservices.util;

import org.apache.commons.net.ntp.NTPUDPClient;
import org.apache.commons.net.ntp.TimeInfo;

import java.io.IOException;
import java.net.InetAddress;

public class NTPUtils {
	
	private static long t = 0;
	
	public static long getTime() throws IOException {
		if(t == 0){
			InetAddress inetAddress = InetAddress.getByName("a.st1.ntp.br");
			
			NTPUDPClient ntpudpClient = new NTPUDPClient();
			TimeInfo timeInfo = ntpudpClient.getTime(inetAddress);
			t = timeInfo.getReturnTime();
		}
		return t;
	}
}

File: src/main/kotlin/br/com/dusty/dservices/util/NTPUtils.kt
package br.com.dusty.dservices.util

import org.apache.commons.net.ntp.NTPUDPClient
import org.apache.commons.net.ntp.TimeInfo

import java.io.IOException
import java.net.InetAddress

object NTPUtils {

    private var t: Long = 0

    val time: Long
        get() {
            if (t == 0L) {
                val inetAddress = InetAddress.getByName("a.st1.ntp.br")

                val ntpudpClient = NTPUDPClient()
                val timeInfo = ntpudpClient.getTime(inetAddress)
                t = timeInfo.returnTime
            }
            
            return t
        }
}
